mlir/include/mlir/Dialect/MIOpen/XdlopsCodeSelection.h

```diff
@@ -436,14 +433,31 @@ struct XdlopsCodeSelection {
         imms.push_back({0, 0, 0});
         argType = VectorType::get({2}, b.getBF16Type());
       } else {
-        llvm::errs() << "Unsupported case:\n";
-        // llvm::errs() << "M, N, K:" << M << " " << N << " " << K << "\n";
-        llvm::errs() << "MPerWave: " << MPerWave << "\n";
-        llvm::errs() << "NPerWave: " << NPerWave << "\n";
-        llvm::errs() << "dataType: ";
-        dataType.dump();
-        llvm::errs() << "\n";
-        llvm_unreachable("Can't meaningfully continue xdlop generation");
+        dumpUnsupported(dataType, MPerWave, NPerWave);
+      }
+    } else if (dataType == b.getIntegerType(8)) {
+      if (MPerWave == 32 && NPerWave == 32) {
+        mfmaInstr = "mfma_i32_32x32x8i8";
+        MPerXdlops = 32;
+        NPerXdlops = 32;
+        MRepeats = 1;
+        NRepeats = 1;
+        vectorType = VectorType::get({16}, b.getI32Type());
+        vectorNumber = 1;
+        imms.push_back({0, 0, 0});
+        argType = VectorType::get({4}, b.getIntegerType(8));
+      } else if (MPerWave == 16 && NPerWave == 16) {
+        mfmaInstr = "mfma_i32_16x16x16i8";
+        MPerXdlops = 16;
+        NPerXdlops = 16;
+        MRepeats = 1;
+        NRepeats = 1;
+        vectorType = VectorType::get({4}, b.getI32Type());
```

Comment:
Hold on a minute, shouldn't this be a vector 16xi8?

Reply:
Hah. I believe vector type is where return type is defined; argType are where argument type are defined.
 - For 16x16x16i8 xdlops, the return type is a vector of 4xi32. 
 - For 32x32x8i8 xdlops, the return type is a vector of 16xi32